WSRequestHandler.cpp

```diff
@@ -1343,6 +1345,287 @@ void WSRequestHandler::HandleSetSceneItemCrop(WSRequestHandler* req) {
+ * @return {int} `crop.bottom` The number of pixels cropped off the bottom of the source before scaling.
+ * @return {int} `crop.left` The number of pixels cropped off the left of the source before scaling.
+ * @return {bool} `visible` If the source is visible.
+ * @return {String} `bounds.type` Type of bounding box.
+ * @return {int} `bounds.alignment` Alignment of the bounding box.
+ * @return {double} `bounds.x` Width of the bounding box.
+ * @return {double} `bounds.y` Height of the bounding box.
+ *
+ * @api requests
+ * @name GetSceneItemSceneProperties
+ * @category sources
+ * @since 4.1.3?
+ */
+void WSRequestHandler::HandleGetSceneItemProperties(WSRequestHandler* req) {
+    if (!req->hasField("item")) {
+        req->SendErrorResponse("missing request parameters");
+        return;
+    }
+
+    const char* item_name = obs_data_get_string(req->data, "item");
+    if (!str_valid(item_name)) {
+        req->SendErrorResponse("invalid request parameters");
+        return;
+    }
+
+    const char* scene_name = obs_data_get_string(req->data, "scene-name");
+    obs_source_t* scene = Utils::GetSceneFromNameOrCurrent(scene_name);
+    if (!scene) {
+        req->SendErrorResponse("requested scene doesn't exist");
+        return;
+    }
+
+    obs_sceneitem_t* scene_item = Utils::GetSceneItemFromName(scene, item_name);
+    if (!scene_item) {
+        req->SendErrorResponse("specified scene item doesn't exist");
+        obs_source_release(scene);
+        return;
+    }
+
+    obs_data_t* data = obs_data_create();
+
+    // is name even needed here?
+    obs_data_set_string(data, "name", item_name);
+
+    obs_data_t* pos_data = obs_data_create();
+    vec2 pos;
+    obs_sceneitem_get_pos(scene_item, &pos);
+    obs_data_set_double(pos_data, "x", pos.x);
+    obs_data_set_double(pos_data, "y", pos.y);
+    obs_data_set_int(pos_data, "alignment", obs_sceneitem_get_alignment(scene_item));
+    obs_data_set_obj(data, "position", pos_data);
+
+    obs_data_set_double(data, "rotation", obs_sceneitem_get_rot(scene_item));
+
+
+    obs_data_t* scale_data = obs_data_create();
+    vec2 scale;
+    obs_sceneitem_get_scale(scene_item, &scale);
+    obs_data_set_double(scale_data, "x", scale.x);
+    obs_data_set_double(scale_data, "y", scale.y);
+    obs_data_set_obj(data, "scale", scale_data);
+
+    obs_data_t* crop_data = obs_data_create();
+    obs_sceneitem_crop crop;
+    obs_sceneitem_get_crop(scene_item, &crop);
+    obs_data_set_int(crop_data, "left", crop.left);
+    obs_data_set_int(crop_data, "top", crop.top);
+    obs_data_set_int(crop_data, "right", crop.right);
+    obs_data_set_int(crop_data, "bottom", crop.bottom);
+    obs_data_set_obj(data, "crop", crop_data);
+
+    obs_data_set_bool(data, "visible", obs_sceneitem_visible(scene_item));
+
+    obs_data_t* bounds_data = obs_data_create();
+    obs_bounds_type bounds_type = obs_sceneitem_get_bounds_type(scene_item);
+    if (bounds_type == OBS_BOUNDS_NONE) {
```

Comment:
Can you guys let me know what you think about the implementation of `bounds type`? The alternative would be to pass an integer instead of a string. Each has its pluses and minuses.
@haganbmj @Palakis

Reply:
Passing an integer in this case is more of a C/C++ trend, so I'd go for the string way here, which I think is more comfortable for JS and Python developers.

Reply:
Bounds type is an enum?
JS would just use whatever the json standard is, which in this case I think is the enum name `OBS_BOUNDS_NONE`

Reply:
`obs_data_set_string(bounds_data, "type", "stretch");`
would become
`obs_data_set_string(bounds_data, "type", "OBS_BOUNDS_STRETCH");`
correct @haganbmj ?

Reply:
@TStod I'd agree with this